cloudify/utils.py

```diff
@@ -266,6 +266,50 @@ def get_execution_token():
         return None
 
 
+def get_instances_of_node(rest_client,
+                          deployment_id=None,
+                          node_id=None,
+                          _ctx=None):
+    """
+    Get node instances of a node from a deployment.
+    """
+    _ctx = _ctx or _get_current_context()
+    deployment_id = deployment_id or _ctx.deployment.id
+    node_id = node_id or _ctx.node.id
+    return rest_client.node_instances.list(deployment_id=deployment_id,
+                                           node_id=node_id)
+
+
+def split_instances_at_index(rest_client,
```

Comment:
this function does A LOT, wouldn't it be better if it was just `def split_instances_at_index(instances, index)`? I don't think this is the best place for actually fetching the node instances